gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/integration/use_case/IngestFederatedApisUseCaseTest.java

```diff
@@ -1476,6 +1482,39 @@ void should_update_doc_if_api_name_changed() {
         }
     }
 
+    @Nested
+    class Metadata {
+
+        @Test
+        void should_update_metadata_if_api_name_changed() {
```

Comment:
I would have put metadata related test cases in Nested `FederatedApiCreation` and `FederatedApiUpdate`

In `FederatedApiCreation`:
- `should_create_api_metadata`

In `FederatedApiUpdate`:
- `should_update_already_ingested_api_metadata`
- `should_not_touch_any_other_metadata`